Documentation/Patterns.md

```diff
@@ -280,3 +280,12 @@ private:
 };
 #endif
 ```
+
+## `type[]` vs. `Array` vs. `Vector` vs. `FixedArray`
```

Comment:
Maybe "`type[]` vs. `Array<type>` vs. `Vector<type>` vs. `FixedArray<type>`" for consistency?